fw/http_parser.h

```diff
@@ -127,7 +127,6 @@ typedef struct {
 		struct {
 			short		dir_flag;
 			char		filled;
-			char		dir_allowed;
 		} cache_control;
 	};
```

Comment:
I finally do not fully understand an idea of the `dir_flag `.
It seems to be unneccessary in request parser (you can make additions (`|=`) directly to `req->cache_ctl.flags` instead of `dir_flag `.
It used in 2 conditions in reqponse parser:
```C
Line 8750: 			if (parser->cache_control.dir_flag == TFW_HTTP_CC_PRIVATE)
Line 8787: 		tokens = parser->cache_control.dir_flag == TFW_HTTP_CC_PRIVATE ?
```
This is because the states `Resp_I_Flag_Maybe_Qal` and `Resp_I_CC_Flag_Argument_Token` are shared between several Cache-Control directives. You can divide these states into several ones to aviod sharing. If you do not like new states... Anyway, it seems, it would be simplier to use the `dir_flag` **only** to cache the directive flag for these checks, not to set `resp->cache_ctl.flags`.

Reply:
Actually there is another condition in the Req_I_CC_Flag/Resp_I_Flag, so we cannot set the req->cache_ctl.flags immediately. However, I agree it's not needed anywhere else and Resp_I_Flag_Maybe_Qal should be segregated into two states.

Reply:
@byko3y 
If you mean
```
WARN_ON_ONCE(!parser->cache_control.dir_flag);
```
this is about internal check for algorithm consistency rather than to parse the text.
However this consistence condition could be proved staticaly.
Is there any way to reach these states without `reg_flag` set?

Reply:
Everything can be proven statically, it's more about how complex the proof is. In this case I do agree the proof is quite simple so there's not much use for this particular statement.